### PR TITLE
Improve connection settings tests

### PR DESCRIPTION
**Pull request by agners**
## Proposed change
This improves mainly the connection settings fixture. The behavior is more closely to real NetworkManager, mainly in that the actual NetworkManager seems to add some default properties automatically (represented by the new `MINIMAL_` settings). Also make the `Update()` call behave largely like NetworkManager, e.g. requiring a minimal set of settings/properties, and update the complete setting but include the `MINIMAL_` setting just like NetworkManager.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:
- Link to client library pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Ruff (`ruff format supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints or add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]
- [ ] [CLI][cli-repository] updated (if necessary)
- [ ] [Client library][client-library-repository] updated (if necessary)

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant
[cli-repository]: https://github.com/home-assistant/cli
[client-library-repository]: https://github.com/home-assistant-libs/python-supervisor-client/

## Summary by CodeRabbit

- **New Features**
	- Enhanced network connection settings management with dynamic fixtures for both Ethernet and wireless connections.
	- Introduced validation for required properties during settings updates to prevent erroneous configurations.
	- Improved handling of IPv4 settings for addresses and DNS, allowing for more robust configuration management.
	- Added new test functionality for Ethernet interface updates to ensure comprehensive testing.

- **Bug Fixes**
	- Adjusted test cases to reflect updated network settings structure, ensuring accurate validation of connection settings.
	- Updated assertions in tests to align with new constant naming conventions, enhancing clarity and maintainability.

### CODE REVIEW

File: tests/dbus_service_mocks/network_connection_settings.py [hunk 1]
```diff
@@ -1,51 +1,31 @@
...
-        "mdns": Variant("i", 2),
-        "permissions": Variant("as", []),
-        "timestamp": Variant("t", 1598125548),
-        "type": Variant("s", "802-3-ethernet"),
-        "uuid": Variant("s", "0c23631e-2118-355c-bbb0-8943229cb0d6"),
-    },
+# NetworkManager Connection settings skeleton which gets generated automatically
+# Created with 1.42.4, using:
+# nmcli con add type ethernet con-name "Test"
+# busctl call org.freedesktop.NetworkManager /org/freedesktop/NetworkManager/Settings/5 org.freedesktop.NetworkManager.Settings.Connection GetSettings --json=pretty
```
Comment (mdegat01) [inline tests/dbus_service_mocks/network_connection_settings.py:18]: I did not know `busctl` could return you output as prettified json. You definitely just saved me a lot of future time jsonifying output from `gdbus` with this 😄

**Review by mdegat01 — APPROVED** [review]
This looks really good and I can see how it mimics network manager much more accurately. Hopefully this saves us some headaches in the future.

On the note of headache, is there any particular tests you want to add around updating network interfaces for edge cases you've seen? Or is updating this good enough to cover the impact of these new properties and ensure we're handling it properly?

**Comment by agners** [comment]
> On the note of headache, is there any particular tests you want to add around updating network interfaces for edge cases you've seen? Or is updating this good enough to cover the impact of these new properties and ensure we're handling it properly? No worries if not, this looks great as is.

Yeah good point. I've added one more API test named `test_api_network_interface_update_ethernet` which changes to "static" then to "auto" and checking what gets written to D-Bus directly, so end-to-end essentially. This tests the current behavior, so we know from what we set out for upcoming changes.

**Review by mdegat01 — APPROVED** [review]
Yea I like the test. Looks good 👍
